### PR TITLE
RT: add one test on sp deactivation

### PR DESCRIPTION
**Pull request by antoine-de**

### CODE REVIEW

**Comment by xlqian** [comment]
hmmm, do you forget to add a exec in cmake file? Never mind, i'll do it

**Comment by xlqian** [comment]
my bad... i just got up from the bed...

**Comment by antoine-de** [comment]
no problem, home working is great :wink:

**Comment by antoine-de** [comment]
you just missed anna's cookies :stuck_out_tongue_winking_eye:

**Comment by xlqian** [comment]
oh, shit...

**Comment by xlqian** [comment]
laisse moi quelques miettes alors!!! hahahaha

File: source/kraken/tests/apply_disruption_test.cpp [hunk 1]
```diff
@@ -109,3 +110,105 @@ BOOST_FIXTURE_TEST_CASE(simple_train_cancellation, SimpleDataset) {
...
+                              *b.data->pt_data, *b.data->meta);
+
+    navitia::apply_disruption(b.impact(nt::RTLevel::Adapted, "S3_closed")
+                              .severity(nt::disruption::Effect::NO_SERVICE)
+                              .on(nt::Type_e::StopPoint, "S3")
+                              .application_periods(btp("20160101T110000"_dt, "20160101T1140001"_dt))
+                              .get_disruption(),
+                              *b.data->pt_data, *b.data->meta);
+
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->vehicle_journeys.size(), 10); //XL: or 8 if you clean vj2_adapted and vj3_adapted
```
Comment (xlqian) [inline source/kraken/tests/apply_disruption_test.cpp:189]: I don't get it why 10, I got 8 vjs even though I don't clean up... Let's talk about when i'm back :)

File: source/kraken/tests/apply_disruption_test.cpp [hunk 2]
```diff
@@ -109,3 +110,105 @@ BOOST_FIXTURE_TEST_CASE(simple_train_cancellation, SimpleDataset) {
...
+
+    auto get_adapted = [&b](const nt::VehicleJourney* vj) {
+        BOOST_REQUIRE_EQUAL(vj->meta_vj->get_adapted_vj().size(), 1); // for our test this is only 1 adapted
```
Comment (xlqian) [inline source/kraken/tests/apply_disruption_test.cpp:192]: I actually get 2 adapted vjs for Vj2. One of them can circulate another doesn't

File: source/kraken/tests/apply_disruption_test.cpp (hunk 1 above)
Comment (xlqian) [inline source/kraken/tests/apply_disruption_test.cpp:189]: oops... you're right...

File: source/kraken/tests/apply_disruption_test.cpp [hunk 3]
```diff
@@ -109,3 +110,105 @@ BOOST_FIXTURE_TEST_CASE(simple_train_cancellation, SimpleDataset) {
...
+BOOST_AUTO_TEST_CASE(multiple_impact_on_stops_different_hours) {
+    ed::builder b("20160101");
+
+    b.sa("S1")("S1");
+    b.sa("S2")("S2");
+    b.sa("S2")("S2");
+
+    const auto* vj1 = b.vj("A").uri("vj1")("S1", "08:00"_t)("S2", "09:00"_t)("S3", "10:00"_t).make();
+    const auto* vj2 = b.vj("A").uri("vj2")("S1", "10:00"_t)("S2", "10:30"_t)("S3", "11:00"_t).make();
+    const auto* vj3 = b.vj("A").uri("vj3")("S1", "10:00"_t)                 ("S3", "10:00"_t).make();
```
Comment (xlqian) [inline source/kraken/tests/apply_disruption_test.cpp:160]: the second stop_point should be "11:00"    O_o

Comment (antoine-de) [inline source/kraken/tests/apply_disruption_test.cpp:160]: yup sorry

File: source/kraken/tests/apply_disruption_test.cpp [hunk 4]
```diff
@@ -109,3 +110,105 @@ BOOST_FIXTURE_TEST_CASE(simple_train_cancellation, SimpleDataset) {
...
+                              .severity(nt::disruption::Effect::NO_SERVICE)
+                              .on(nt::Type_e::StopPoint, "S2")
+                              .application_periods(btp("20160101T100000"_dt, "20160101T1120001"_dt))
+                              .get_disruption(),
+                              *b.data->pt_data, *b.data->meta);
+
+    navitia::apply_disruption(b.impact(nt::RTLevel::Adapted, "S3_closed")
+                              .severity(nt::disruption::Effect::NO_SERVICE)
+                              .on(nt::Type_e::StopPoint, "S3")
+                              .application_periods(btp("20160101T110000"_dt, "20160101T1140001"_dt))
```
Comment (xlqian) [inline source/kraken/tests/apply_disruption_test.cpp:185]: disruption time is not correct either

File: source/kraken/tests/apply_disruption_test.cpp [hunk 5]
```diff
@@ -109,3 +110,105 @@ BOOST_FIXTURE_TEST_CASE(simple_train_cancellation, SimpleDataset) {
...
+    //no adapted for vj1
+    BOOST_REQUIRE_EQUAL(vj1->meta_vj->get_adapted_vj().size(), 0);
+
+    //TODO XL, cf comment at the begining of the test for vj2 and vj3
+    const auto* vj2_adapted = get_adapted(vj2);
+    BOOST_CHECK(ba::ends_with(vj2->base_validity_pattern()->days.to_string(), "111"));
+    BOOST_CHECK(ba::ends_with(vj2->adapted_validity_pattern()->days.to_string(), "110"));
+    BOOST_CHECK(ba::ends_with(vj2_adapted->base_validity_pattern()->days.to_string(), "000"));
+    BOOST_CHECK(ba::ends_with(vj2_adapted->adapted_validity_pattern()->days.to_string(), "001"));
+    check_vj(vj2_adapted, {nt::StopTime{"08:00"_t, "08:00"_t, s1}});
```
Comment (xlqian) [inline source/kraken/tests/apply_disruption_test.cpp:205]: "10:00"

File: source/kraken/tests/apply_disruption_test.cpp [hunk 6]
```diff
@@ -109,3 +110,105 @@ BOOST_FIXTURE_TEST_CASE(simple_train_cancellation, SimpleDataset) {
...
+
+    check_vj(get_adapted(vj3), {{"08:00"_t, "08:00"_t, s1}});
```
Comment (xlqian) [inline source/kraken/tests/apply_disruption_test.cpp:207]: "10:00"

**Comment by xlqian** [comment]
Thx for the test :D

**Pull request merged**